pkg/operator/starter.go

```diff
@@ -117,23 +121,41 @@ func RunOperator(ctx context.Context, controllerContext *controllercmd.Controlle
 	// we create a new informer directly because we are only interested in observing changes to the master machines
 	// primarily to avoid reconciling on every update in large clusters (~2K machines)
 	masterMachineInformer := machineinformersv1beta1.NewFilteredMachineInformer(machineClientSet, "openshift-machine-api", 1*time.Hour, cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc}, func(listOptions *metav1.ListOptions) {
-		listOptions.LabelSelector = masterMachineLabelSelectorString
+		listOptions.LabelSelector = fmt.Sprintf("%s in (%s,%s)", masterMachineLabelSelectorKeyString, masterMachineLabelSelectorValueString, arbiterMachineLabelSelectorValueString)
 	})
-	masterMachineLabelSelector, err := labels.Parse(masterMachineLabelSelectorString)
+	masterMachineLabelSelector, err := labels.Parse(fmt.Sprintf("%s in (%s,%s)", masterMachineLabelSelectorKeyString, masterMachineLabelSelectorValueString, arbiterMachineLabelSelectorValueString))
 	if err != nil {
 		return err
 	}
+
+	arbiterNodeLabelSelector, err := labels.Parse(arbiterNodeLabelSelectorString)
+	if err != nil {
+		return err
+	}
+
 	// we create a new informer directly because we are only interested in observing changes to the master nodes
 	// primarily to avoid reconciling on every update in large clusters (~2K nodes)
-	masterNodeInformer := corev1informers.NewFilteredNodeInformer(kubeClient, 1*time.Hour, cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc}, func(listOptions *metav1.ListOptions) {
+	controlPlaneNodeInformer := corev1informers.NewFilteredNodeInformer(kubeClient, 1*time.Hour, cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc}, func(listOptions *metav1.ListOptions) {
 		listOptions.LabelSelector = masterNodeLabelSelectorString
 	})
-	masterNodeLister := corev1listers.NewNodeLister(masterNodeInformer.GetIndexer())
-	masterNodeLabelSelector, err := labels.Parse(masterNodeLabelSelectorString)
+	controlPlaneNodeLister := corev1listers.NewNodeLister(controlPlaneNodeInformer.GetIndexer())
+	controlPlaneNodeLabelSelector, err := labels.Parse(masterNodeLabelSelectorString)
 	if err != nil {
 		return err
 	}
 
+	var hasArbiterTopology bool
+	if infra, err := configClient.ConfigV1().Infrastructures().Get(ctx, ceohelpers.InfrastructureClusterName, metav1.GetOptions{}); err != nil {
+		return err
+	} else {
+		hasArbiterTopology = infra.Status.ControlPlaneTopology == configv1.HighlyAvailableArbiterMode
+	}
```

Comment:
can we move this into one of our helpers? 

we have https://github.com/openshift/cluster-etcd-operator/blob/master/pkg/operator/ceohelpers/control_plane_topology.go#L26 for SNO already